carbonmark-api/src/utils/marketplace.utils.ts

```diff
@@ -32,6 +34,17 @@ export type GetProjectListing = NonNullable<
   GetProjectsQuery["projects"][number]["listings"]
 >[number];
 
+/** Format amounts or quantities by registry decimals */
+/** Currently all registries use 18 decimals except ICR, which uses 0  */
+
+export const formatAmountByRegistry = (registry: string, quantity: string) => {
+  if (registry === REGISTRIES.ICR.id) {
+    return formatUnits(quantity, REGISTRIES.ICR.decimals);
+  } else {
+    return formatUnits(quantity, 18);
+  }
+};
+
```

Comment:
This seems a little unintuitive to me. Can we do something like this instead?

```suggestion
export const formatAmountByRegistry = (registryId: string, quantity: string) => {
  return formatUnits(quantity, REGISTRIES[registryId].decimals);
};
```

When we later add another registry, we won't have to update this function. We just need to add the registry with it's decimals to `REGISTRIES`.

Reply:
Good call, revised